src/rt/sched/backpressure.h

```diff
@@ -0,0 +1,249 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+/**
+ * ## Introduction
+ *
+ * The primary goal of backpressure is to prevent runaway growth of a cown's
+ * message queue that might eventually result in a Verona application running
+ * out of memory. Cowns that receive messages at a higher rate than they are
+ * able to process are considered "overloaded" and may lead to this unbounded
+ * queue growth. Once an overloaded cown is identified, cowns sending messages
+ * to it will be temporarily descheduled ("muted") so that the overloaded cown
+ * can work through its queue. Similarly, cowns sending to muted receivers will
+ * also be muted so that cowns do not themselves become overloaded as a
+ * consequence of being muted. Muted cowns will be rescheduled after the
+ * receiver that triggered their muting (the "mutor") is no longer muted or
+ * overloaded.
+ *
+ * ## Detecting Overloaded Cowns
```

Comment:
This section assumes that someone knows or would otherwise reach for measuring queue depth as a means of knowing that something is overloaded. That seems reasonable to me, however, I'm not sure it really would hold.

The problem as I see it is that we just threw a ton of information at someone in the previous sentence and they might not have absorbed it all. It's very dense and there is a lot to cover.

I think it would be better to be safe, and start with something about how we need to determine overload and that while the basic definition (unbounded queue growth) is straightforward, the implementation isn't.

Avoiding "unbounded queue growth" is important but because there are so many cowns, really we need a heuristic that will allow us to provide an arbitrary bounding by saying that a cown is overloaded.

One method of doing this would be count the number of items in the queue on a regular basis and compare those lengths over time to determine if it is growing in a fashion that looks unbounded.  However, counting every item in every queue for every cown adds a tremendous amount of overhead to the runtime and is thus avoided.

The current means of determining if a cown is overloaded is to use the following heuristic: